plugins/MachineSettingsAction/MachineSettingsAction.qml

```diff
@@ -221,44 +183,13 @@ Cura.MachineAction
                                 }
 
-                                ComboBox
+                                Loader
                                 {
-                                    model: ListModel
-                                    {
-                                        id: flavorModel
-                                        Component.onCompleted:
-                                        {
-                                            // Options come in as a string-representation of an OrderedDict
-                                            var options = machineGCodeFlavorProvider.properties.options.match(/^OrderedDict\(\[\((.*)\)\]\)$/);
-                                            if(options)
-                                            {
-                                                options = options[1].split("), (")
-                                                for(var i = 0; i < options.length; i++)
-                                                {
-                                                    var option = options[i].substring(1, options[i].length - 1).split("', '")
-                                                    flavorModel.append({text: option[1], value: option[0]});
-                                                }
-                                            }
-                                        }
-                                    }
-                                    currentIndex:
-                                    {
-                                        var currentValue = machineGCodeFlavorProvider.properties.value;
-                                        var index = 0;
-                                        for(var i = 0; i < flavorModel.count; i++)
-                                        {
-                                            if(flavorModel.get(i).value == currentValue) {
-                                                index = i;
-                                                break;
-                                            }
-                                        }
-                                        return index
-                                    }
-                                    onActivated:
-                                    {
-                                        machineGCodeFlavorProvider.setPropertyValue("value", flavorModel.get(index).value);
-                                        manager.updateHasMaterialsMetadata();
-                                    }
+                                    id: gcodeFlavorComboBox
+                                    sourceComponent: comboBoxWithOptions
+                                    property string settingKey: "machine_gcode_flavor"
+                                    property bool forceUpdateOnChange: true
+                                    property string afterOnActivate: "manager.updateHasMaterialsMetadata()"
```

Comment:
Is this really the only way to do this? Dynamic evals scare me if the GUI needs to set them.